app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -0,0 +1,250 @@
+
+                Bitmap bitmapImage = BitmapFactory.decodeByteArray(photo, 0, photo.length, null);
+                //Image rotation
+
+                if(orientationResult != null)
+                {
+                    switch (orientationResult)
+                    {
+                        case "LR": bitmapImage=rotateImage(bitmapImage,90); break;
+                        case "LL": bitmapImage=rotateImage(bitmapImage,270); break;
+                        case "PU": bitmapImage=rotateImage(bitmapImage,180); break;
+                        default: break;
+                    }
+
+                    //Temporary stores the captured photo into a file that will be used from the Camera Result activity
+                    String filePath= tempFileImage(CameraActivity.this, bitmapImage,"capturedImage");
+
+                    SharedPreferences prefs = getSharedPreferences("prefs", MODE_PRIVATE);
+                    SharedPreferences.Editor edit = prefs.edit();
+                    edit.putString("imagePath", filePath.trim());
+                    edit.apply();
+
+                    //An intent that will launch the activity that will analyse the photo
+                    Intent i = new Intent(CameraActivity.this, ResultActivity.class);
+                    startActivity(i);
+                }
+            }
+        });
+
+    }
+
+
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        cameraKitView.onStart();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        cameraKitView.onResume();
+    }
+
+    @Override
+    protected void onPause() {
+        cameraKitView.onPause();
+        super.onPause();
+    }
+
+    @Override
+    protected void onStop() {
+        cameraKitView.onStop();
+        super.onStop();
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        cameraKitView.onRequestPermissionsResult(requestCode, permissions, grantResults);
+    }
+
+
+    /**
+     * Stores the captured image into a temporary file useful to pass large data between activities
+     * and returns the file's path.
+     * @param context The reference of the current activity
+     * @param bitmap The captured image to store into the file. Not null or empty.
+     * @param name The name of the file. Not null or empty.
+     * @return The files path
+     * @author Leonardo Rossi
+     */
+    private String tempFileImage(Context context, Bitmap bitmap, String name)
+    {
+
+        File outputDir = context.getCacheDir();
+        File imageFile = new File(outputDir, name + ".jpg");
+
+        OutputStream os;
+        try {
+            os = new FileOutputStream(imageFile);
+            bitmap.compress(Bitmap.CompressFormat.JPEG, 100, os);
+            os.flush();
+            os.close();
+        } catch (Exception e) {
```

Comment:
Should be better to not catch every Exception but only the one referred to OutputStream

Reply:
Thanks for the suggestion but at first glance this could be a great observation but in this case I have to leave as it is. The problem is that I could change "Exception" to "FileNotFoundException" but os.flush(); and os.close(); require "Exception" as catch so if I want to include them into the try catch I have to leave "Exception". Anyway "Exception" also trigger "FileNotFoundException"

Reply:
ok, thank for the explanation